src/foam/nanos/servlet/VirtualHostRoutingServlet.js

```diff
@@ -46,6 +59,98 @@ foam.CLASS({
       javaCode: 'setServletConfig(config);',
       code: function() { }
     },
+    {
+      name: 'getTagString',
+      type: 'String',
+      documentation: `Utility method that returns an html tag string.`,
+      args: [ 
+        { name: 'tag', javaType: 'String'},
+        { name: 'keywords', javaType: 'java.util.ArrayList<String>'},
+        { name: 'attributes', javaType: 'java.util.HashMap'},
+        { name: 'hasClosingTag', javaType: 'Boolean'},
+        { name: 'isSingleClosingTag', javaType: 'Boolean'}
+      ],
+      javaCode: `
+        String tagString = "<" + tag + " ";
+        for ( String keyword : keywords ) {
+          tagString += keyword + " ";
+        }
+        java.util.Iterator i = attributes.entrySet().iterator();
+        while ( i.hasNext() ) {
+          java.util.Map.Entry attribute = (java.util.Map.Entry) i.next();
+          tagString += attribute.getKey() + "=\\"" + attribute.getValue() + "\\" ";
+          i.remove();
+        }
+        if ( isSingleClosingTag ) {
+          return tagString += "/>";
+        }
+        tagString += ">";
+        tagString += hasClosingTag ? "</" + tag + ">" : "";
+        return tagString;
+      `
+    },
+    {
+      name: 'populateHead',
+      type: 'Void',
+      documentation: `
+        Generates the index file's head content based on theme and prints it to the response writer.
+      `,
+      args: [ 
+        { name: 'theme', javaType: 'foam.nanos.theme.Theme'},
+        { name: 'out', javaType: 'java.io.PrintWriter'} 
+      ],
+      javaCode: `
+      out.println("<meta charset=\\"utf-8\\"/>");
+      out.println("<title>" + theme.getAppName() + "</title>");
+      java.util.ArrayList<java.util.HashMap> headConfig = (java.util.ArrayList<java.util.HashMap>) theme.getHeadConfig();
+
+      for ( int i = 0; i < headConfig.size(); i++ ) {
+        java.util.HashMap tagConfig = (java.util.HashMap) headConfig.get(i);
+        java.util.ArrayList<String> keywords = new java.util.ArrayList<String>();
+        if ( tagConfig.containsKey("keywords") ) {
+          keywords = (java.util.ArrayList<String>) tagConfig.get("keywords");
+        }
+        Boolean hasClosingTag = false;
+        if ( tagConfig.containsKey("hasClosingTag") ) {
+          hasClosingTag = (Boolean) tagConfig.get("hasClosingTag");
+        }
+        Boolean isSingleClosingTag = false;
+        if ( tagConfig.containsKey("isSingleClosingTag") ) {
+          isSingleClosingTag = (Boolean) tagConfig.get("isSingleClosingTag");
+        }
+        String headTag = this.getTagString((String)tagConfig.get("tag"), keywords, (java.util.HashMap)tagConfig.get("attributes"), hasClosingTag, isSingleClosingTag);
+        out.println(headTag);
+      }
+      if ( this.getIsProduction() ) {
```

Comment:
So this test is not for production, but rather if -u on the build.  The short src path and VERSION is needed when we build to jar file. 
Perhaps we can a JVM flag to indicate how the application is executing - from jar or filesystem.